server/Dockerfile

```diff
@@ -77,7 +77,6 @@ COPY --from=builder /usr/src/kanidm/server/core/static /hpkg
 RUN chmod +x /sbin/kanidmd
 
 EXPOSE 8443 3636
-VOLUME /data
```

Comment:
> could you please update the broken link

Link to source file and line and sure 👍 

> as the folder doesn't exist

Would it make sense to default to the location instead of `/` when running the `ENTRYPOINT`?

**EDIT:** Oh you only have `CMD` set (`/sbin/kanidmd server -c /data/server.toml`), that is still absolute path so a `WORKDIR` directive would create the location (_not sure if that avoids an extra layer or not, I think it might_).

Alternatively, I could add `&& mkdir /data` in an earlier `RUN`?

Reply:
The `WORKDIR` doesn't apply to the final container, note the `FROM repos` on line #65.

> Alternatively, I could add && mkdir /data in an earlier RUN?

as long as it's after line #65

>> could you please update the broken link
> Link to source file and line and sure 👍

https://github.com/search?q=repo%3Akanidm%2Fkanidm+%2FTo+change+the+origin+or+domain+name+see%2F+language%3ARust&type=code

Reply:
> The `WORKDIR` doesn't apply to the final container, note the `FROM repos` on line 65.

Sorry I don't follow? That is the last `FROM` directive in this file, and given the remainder of the file it looks like the final stage? Are you doing something else to the image?

Separate `Dockerfile`:

```Dockerfile
FROM alpine
WORKDIR /data
```

`compose.yaml` that extends that image with a custom `Dockerfile`:

```yaml
services:
  example:
    # The `image` setting now represents the tag for the local build configured below:
    image: local/workdir-extended:latest
    # Local build (no need to try pull `image` remotely):
    pull_policy: build
    # The original image was extended, the directory from WORKDIR still exists.
    # NOTE: If this were `FROM scratch`, obviously you'd need to COPY `/data` over
    #       but that's no different to how you approach creating `/data` prior.
    build:
      dockerfile_inline: |
        FROM local/workdir:latest
        RUN touch hello.txt
        CMD pwd && ls
```

```console
# Build the separate image:
$ docker build --tag local/workdir .

# Build and run the variant image that extends it:
$ docker compose run --rm example
/data
hello.txt

# Also verify image metadata still has the workdir set:
$ docker image inspect local/workdir-extended:latest | jq -r '.[].Config.WorkingDir'
/data
```

Seems to work perfectly fine no?

Reply:
> > > could you please update the broken link
> > > Link to source file and line and sure 👍
> 
> https://github.com/search?q=repo%3Akanidm%2Fkanidm+%2FTo+change+the+origin+or+domain+name+see%2F+language%3ARust&type=code

That is not the line you wanted changed? 🤷‍♂️ 

I've used that to identify the intended line and correct the link with the one you've provided, so thanks 👍

Reply:
I thought you were talking about the _existing_ workdir command, not adding one.

Reply:
> the _existing_ workdir command, not adding one.

That's in the builder stage and wouldn't make sense to replace an existing one with a different path.

- If it makes sense to run the container from `/data` I can add a `WORKDIR` directive. I assumed `/data` was arbitrary for documentation, and not hard-coded/expected anywhere.
- If that assumption is correct, you could just use an existing directory for documentation and anyone who wants to use a custom location like `/data` can do so via config + volume?

How would you like to proceed?
- `WORKDIR` or `RUN`?
- `/data` or an existing location like `/tmp`?